Dataset

DavidFM43 opened the pull request:
- `train.py`: Refactor
	 - Quitar instalación de `pytorch-segmentation-models` ahora se hace desde el notebook de kaggle.
	 - Nombrar mejor los argumentos cuando se llama el script por CLI.
	 - Añadir las transformaciones de los datasets:
		  - Normalización de los canales de las imagenes.
		  - Antes el dataset retornaba directamente los tensores, ahora eso se maneja con los transforms de torchvision, creo que queda mejor así pero no tengo una razón muy fuerte tampoco.
	 - Random split del dataset en entrenamiento, validación y testeo. La parte de testeo realmente no se usa por lo que es el script de entrenamiento.
- `README.md`: 
	- Añadir sección "Setup" que explica como descargar/procesar los datos, correr el script en entrenamiento en Kaggle y loggear en wandb.
	 - Quitar la sección de TODOs.
- `dataset.py`: Reformat del script, ahora es mas simple.
- `utils.py`: Añadir función para revertir la normalización y obtener imagen original. Mejorar algún docstring.
- `train_kaggle.ipynb`: Notebook para correr el entrenamiento en Kaggle. Remplaza el anterior script `train_kaggle.py`.
- `get_key.py`: Script para configurar la API key de wandb dependiendo si se corre localmente o en Kaggle.
- `prepare.py`: Script para descargar brutos y procesarlos localmente.
- Reorganización de la carpeta `data` en una estructura mas clara y sencilla, se especifica en el `README.md`.
- `requirements.txt`: Actulizar librerias y quitar `torch` y `torchvision` pues la instalación depende del hardware en que se corra.
- `config.yaml`: Antes se llamaba `1.yaml`; Refactor pequeños por estetica.
- `.gitignore`: Añadir el script `key.py` que contiene la llave de wandb.
- `notes/`: Accidentalmente añadí cosas de mas de Obsidian.

DavidFM43 committed:
Reformat and improve data download

- Add `prepare.py` script that downloads and prepares the dataset
- Simplify the `dataset.py` script

DavidFM43 committed:
Removed data-preprocessing notebook

DavidFM43 committed:
Update notes

DavidFM43 committed:
Small changes in dataset.py

DavidFM43 committed:
Set wandb key depending on running enviroment

DavidFM43 committed:
Refactor train script

DavidFM43 committed:
Fix type of lr

DavidFM43 committed:
Fix type and dims in targets in dataset

DavidFM43 committed:
Change loss fn to CE

DavidFM43 committed:
Remove softmax from model config

DavidFM43 committed:
Add logging variable as cli argument

DavidFM43 committed:
Rename log cli argument

DavidFM43 committed:
Fix scaling bug in labels

DavidFM43 committed:
Sort import in dataset.py

DavidFM43 committed:
Add undo normalization transformation

DavidFM43 committed:
Update requirements.txt

DavidFM43 committed:
Update README.md

eguar11011 commented on train.py:
```diff
@@ -107,7 +110,7 @@
 
-# ij is the number of pixels of class i predicted to belong to class j
+# columns are the predictions and rows are the real labels
 conf_matrix = torch.zeros((7, 7), device=device)
```
Debería haber una explicación de como se esta configurando la matriz.

DavidFM43 replied:
Es la matriz de confusión, a que se refiere con la configuración de la matriz?

eguar11011 replied:
No dice que esa es la matriz de confusión, entonces yo pensaría que colocar ese comentario de que esa es la matriz de confusión.

DavidFM43 replied:
Ese comentario ya esta

eguar11011 commented on utils.py:
```diff
@@ -1,9 +1,17 @@
 from tqdm import tqdm
 
 
+class UnNormalize(torchvision.transforms.Normalize):
```
Sería bueno dejar un link de porque esto de normalización es bueno. He visto que en algunos repositorios hacen eso,

DavidFM43 replied:
Hmmm pues creo que es una practica común, en especial cuando usa modelos pre-entrenados, pero no sé si hay un lugar donde expliquen al detalle, usted que ha visto?

DavidFM43 replied:
Pues la verdad no lo veo necesario, pero si quiere lo puede poner usted

eguar11011 replied:
Sí, algo así como hacen en este repo: https://github.com/wolny/pytorch-3dunet/blob/master/pytorch3dunet/unet3d/losses.py

DavidFM43 replied:
Si pero en ese caso es porque esa idea proviene de un paper, pero creo que normalizar no tiene un autor o una fuente en especifico.

DavidFM43 replied:
Yo creo que ponerlo en un todo y si algo se hace despues

eguar11011 replied:
Listo. Yo creo que mejor despues

eguar11011 commented on train.py:
```diff
@@ -54,7 +53,7 @@
     [
         transforms.Resize(resize_res),
-        transforms.ToTensor(),
+        transforms.PILToTensor(),
```
En el commit creo que debería explicar cual era el error específicamente, porqué no había ningún Issues así que no sé a cual error se refiere.

DavidFM43 replied:
Pues el commit creo que se llama "fix scaling bug in labels", osea "corrigiendo bug de escalado en las etiquetas", básicamente es porque la función anterior escalaba los valores pero pues para las etiquetas no se quiere que se modifiquen, entonces ese es el bug. Supongo que podria explicar mejor esto en el commmit, pero tocaria hacer force-push y creo que eso no es muy bueno. Puedo mejor especificar en la descripcion de la PR.

eguar11011 replied:
Si me parece buena idea.

DavidFM43 replied:
Ya esta actualizado la descripción de la PR

eguar11011 commented on notes/.obsidian/workspace.json:
```diff
@@ -4,16 +4,16 @@
     "children": [
       {
-        "id": "3fc41fefca8ecc80",
+        "id": "559065886dc4a506",
```
¿estos cambios de que son? 

Creo que deberíamos colocar en gitignore estos archivos json de obsidian.

DavidFM43 replied:
Ah esque se me fue unas cosas del obsidian por accidente xd, la verdad no se como manejar esto mejor

DavidFM43 replied:
Como asi colocarlos en obsidian?

eguar11011 replied:
En el gitignore, los archivos json de obsidian.

DavidFM43 replied:
Listo

eguar11011 commented on train.py:
```diff
@@ -1,115 +1,93 @@
             optimizer=type(optimizer).__name__,
-            loss_fn="DiceLoss",
+            # TODO: Automatically put loss fn name
+            loss_fn="dice loss",
```
Aquí falta cambia CE.

DavidFM43 replied:
Esto esta desactualizado, ese cambio ya se hizó

eguar11011 commented on get_key.py:
```diff
@@ -0,0 +1,9 @@
+import pkgutil
```
Creo que dejar un link para buscar sobre esto sería bueno.

DavidFM43 replied:
Puede ser mas especifico?

eguar11011 replied:
Si claro. Pensaba en dejar un link de la documentación de eso.

eguar11011 replied:
Aunque no lo veo necesario. Podemos dejarlo así.

eguar11011 commented on notebooks/2.0-Data-preprocessing.ipynb:
```diff
@@ -1,102 +0,0 @@
-{
```
¿Por qué se quito el notebook del preprocesamiento? ya que sería bueno por si alguien quiere revisar como se hizo el preprocesamiento-

DavidFM43 replied:
Porque el ese notebook estaba desactualizado y puede revisar el preprocesamiento en el script `prepare.py`

eguar11011 replied:
Aaah listo.

eguar11011 commented on README.md:
```diff
@@ -1,9 +1,52 @@
+
+### Running the training script in kaggle
+
+In case you down have a machine with GPU, which is highly needed at this point, you can optionally run the training script on kaggle and user their free GPUs. To do that simply go to the `train_kaggle.ipynb` notebook and click on the botton "Run in kaggle".
```
Ese notebook `train_kaggle.ipynb`no existe.

DavidFM43 replied:
Ya existe

eguar11011 commented on train.py:
```diff
@@ -1,115 +1,96 @@
 if wandb_log:
-    wandb.login(key="2699e8522063dc2ad0f359c8230e5cc09db3ebd8")
+    wandb.login(key=wandb_key)
     # TODO: Add entity parameter to log to the team account
```
¿A cual parámetro se refiere?

DavidFM43 replied:
Ah si ese comentario ya no es necesario

eguar11011 commented on utils.py:
```diff
@@ -1,7 +1,15 @@
+
+
+class UnNormalize(torchvision.transforms.Normalize):
+    def __init__(self, mean, std, *args, **kwargs):
```
Falta agregar el docstring

DavidFM43 replied:
Listo

DavidFM43 committed:
Update README.md

DavidFM43 committed:
Add the new train_kaggle notebook and remove old script

Add new train_kaggle notebook to run the training script in kaggle and
remove the old script.

DavidFM43 committed:
Add comment of confusion matrix

DavidFM43 committed:
Fix typo

Commit:
ignorar los archivos json de obsidian

DavidFM43 committed:
Remove outdated comment

DavidFM43 committed:
Add docstring to function

DavidFM43 committed:
Merge remote-tracking branch 'origin/dataset' into dataset

DavidFM43 committed:
Clean notes folder

Commit:
Agregar el paper1 para la discusion de este mismo
